src/pocketdb/consensus/Social.h

```diff
@@ -78,8 +78,9 @@ namespace PocketConsensus
         // Generic transactions validating
         virtual ConsensusValidateResult Check(const CTransactionRef& tx, const shared_ptr<T>& ptx)
         {
-            if (auto[ok, result] = CheckOpReturnHash(tx, ptx); !ok)
-                return {false, result};
+            if (auto[ok, result] = CheckAdPost(tx, ptx); !ok)
+                if (auto[ok, result] = CheckOpReturnHash(tx, ptx); !ok)
+                    return {false, result};
```

Comment:
Maybe it's a typo? I would suggest moving the CheckAdPost inside the AdPost model?

Reply:
I'll try to find a cleaner way to do this, but this is to check the transaction which accepts the adpost rather than the original adpost itself.